### DISCUSSION TITLE
The uSD tests appear to have FAILED.

### DISCUSSION THREAD

**Discussion by it-mauri** [thehighnibble/thehighnibble.github.io#113] [IMSAI8080esp replica]
I have been using my IMSAI8080esp for a while. Then today, after few weeks of rest, I have started again the machine and can't see the WiFi network.
By checking the terminal of the ESP this is all I got:

```
rst:0x1 (POWERON_RESET),boot:0x3f (SPI_FAST_FLASH_BOOT)
configsip: 188777542, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0018,len:4
load:0x3fff001c,len:4492
load:0x40078000,len:13324
ho 0 tail 12 room 4
load:0x40080400,len:3796
entry 0x40080624
I (354) psram: This chip is ESP32-PICO
I (354) spiram: Found 64MBit SPI RAM device
I (354) spiram: SPI RAM mode: flash 80m sram 80m
I (357) spiram: PSRAM initialized, cache is in low/high (2-core) mode.
I (364) cpu_start: Pro cpu up.
I (368) cpu_start: Application information:
I (373) cpu_start: Project name:     imsaisim_esp32
I (378) cpu_start: App version:      v1.9.1
I (383) cpu_start: Compile time:     Sep  8 2021 15:28:40
I (389) cpu_start: ELF file SHA256:  8a0e39da91675446...
I (395) cpu_start: ESP-IDF:          v4.0.2-4-g877a28e05
I (401) cpu_start: Starting app cpu, entry point is 0x40081b58
I (0) cpu_start: App cpu up.
I (898) spiram: SPI SRAM memory test OK
I (962) heap_init: Initializing. RAM available for dynamic allocation:
I (962) heap_init: At 3FFAE6E0 len 00001920 (6 KiB): DRAM
I (964) heap_init: At 3FFC36F8 len 0001C908 (114 KiB): DRAM
I (970) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM
I (976) heap_init: At 3FFE4350 len 0001BCB0 (111 KiB): D/IRAM
I (983) heap_init: At 4009D62C len 000029D4 (10 KiB): IRAM
I (989) cpu_start: Pro cpu start user code
I (994) spiram: Adding pool of 3241K of external SPI memory to heap allocator
I (1014) spi_flash: detected chip: gd
I (1014) spi_flash: flash io: qio
I (1015) cpu_start: Starting scheduler on PRO CPU.
I (0) cpu_start: Starting scheduler on APP CPU.
I (1084) nvs: Reading settings from NVS 
I (1085) nvs: settings = 00000450
I (1085) esp32_boot: Log Level set to NONE
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
! The uSD tests appear to have FAILED.
!
! Please take note of the above message(s) and review the state of the
! uSD card and/or your soldering.
!
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

Does this mean I am experienced a kind of hardware issue?
Thank for any kind help.
Alessandro

**Comment by nbreeden2** [comment]
Can you read the uSD card on your PC? Is it formatted and has the files/folder structure on it?

**Comment by it-mauri** [comment]
I think so, yep !
<img width="211" alt="Screenshot 2023-10-13 alle 22 27 45" src="https://github.com/thehighnibble/thehighnibble.github.io/assets/147879940/c939d4bd-4bfe-4e51-b615-aa11e7a9eb15">

**Comment by it-mauri** [comment]
Well. I just unplugged the SD card to move it to the computer, then back to the IMSAI... paying much attention while inserting the uSD card again, and now... puff, all is working again ^_^

**Comment by it-mauri** [comment]
All is very unstable. Sometimes the uSD gets read, others not. And sometimes asking for "DIR" needs many seconds to get the results, other times is fast as it should. I suspect the uSD is damaged somehow.
How to write a new uSD with the whole sw?

**Reply by nbreeden2** [reply]
You'll find the image (IMAGE.zip) and the instructions at the URL below. Pay attention to the version of firmware you're running, read the warnings on the page.

Basically, it's this:
Copy all the contents of the SD card to a backup folder on your PC.
Format the SD Card using FAT32
Unzip the contents of IMAGE.zip onto the SD card.
You'll end up with 3 root folders on the card:

cromemco
imsai
www

Leave all 3 folders in place. 
cromemco can be left on the card. Should you decide to run the cromemco emulation instead of the imsai this is the folder it will use, there's no harm in leaving it in place.

https://github.com/thehighnibble/firmware/releases

**Comment by willjlong** [comment]
I had something similar.  I got an SD card extender and checked continuity.  After reflowing the SD socket soldering 4 times, the SD socket became stable.
